### DISCUSSION TITLE
Is it possible to add a columnseries to a candlestick series chart?

### DISCUSSION THREAD

**Discussion by preintel** [Live-Charts/LiveCharts2#208] [Q&A]
Will have the candlesticks in the chart but want to also add a columnseries using the same x axis to indicate the amount of volume during each candlestick. Seems to throw an error when adding the column series to the existing candlestick series.

**Comment by beto-rodriguez** [comment]
Please show your code, and the exception message you are getting, it should be possible.

**Comment by CodeWithMichal** [comment]
What does it mean that you are adding column series to the existing candlestkick series? You need to create separate serie and add both of them to CartesianChart Series property

**Comment by preintel** [comment]
Is there an easy way to increase the margins of the max/min values of a y axis? I was looking in the axis class but could not figure it out.
Basically if I have a min value of 10 and a max value of 100, can I add a 10% margin so the graph shows the y values from 9 to 110?

**Reply by CodeWithMichal** [reply]
As fair as I know you have to do it yourself. If you have live/dynamic data you could find min and max values from your serie and based on that set axis limits. And this is something you probably want to do every time you update your data

**Reply by preintel** [reply]
That is what I am doing now and it works but was hoping the Axis class would have an option for it instead.
I was wondering if there were themes that could be easily applied to the cartesian charts in Winforms version, I saw that WPF supported these but figure WPF is more versatile than winforms in this regard.

**Comment by preintel** [comment]
If I am using this code to create the cartesianChart, what type of point do I use for the Bar/column series to map the x to DateTime and y to a double number? The candlestickseries appears fine but when I add the series ColumnSeries<ObservablePoint> list which has a series of ObservablePoints (double, double), it throws an exception at (X axis line: Labeler = value => new DateTime((long)value).ToString("hh:mm tt")) as I am unable to pass the DateTime as a double through ObservablePoint even when using double.ToOADate() and passing that as the X variable it throws an error. I get the same error when I use ColumnSeries<DateTimePoint> passing a set of DateTimePoints(DateTime, double) for the column which shares the same axis (DateTime) as the candlestick series and I am passing the same Datetime variable to both series however for financial I am passing the datetime via (FinancialPoint(DateTime, (double)highPrice, (double)openPrice, (double)closePrice, (double)lowPrice).

```
ViewModel viewM = new ViewModel();
                            viewM.Series = new ObservableCollection<ISeries>
                                                     {
                                                          new CandlesticksSeries<FinancialPoint>
                                                         {
                                                              Name = ticker,
                                                             Values = fPList
                                                         }, new ColumnSeries<ObservablePoint>
                                                         {
                                                             Name = "test",
                                                             Values = //ObservableCollection<ObservablePoint> 
                                                         }
                                                    };

                            viewM.XAxes = new[]
                                 {
                                                                    new Axis
                                                                    {
                                                                        Name = "Date: "+DateTime.Now.Date.ToString("MM/dd/yyyy"),
                                                                        LabelsRotation = 15,
                                                                        Labeler = value => new DateTime((long)value).ToString("hh:mm tt"),
                                                                        UnitWidth = TimeSpan.FromMinutes(1).Ticks
                                                                    }
                                                                };

                            viewM.YAxes = new[]
                                      {
                                                                    new Axis
                                                                    {
                                                                        Name = "Price",
                                                                        Labeler = value => "$"+value.ToString(),
                                                                        MaxLimit = maxPrice + (0.1*maxPrice),
                                                                        MinLimit = minPrice - (0.1*minPrice)
                                                                        //,UnitWidth = 1
                                                                    }
                                                                };
                            cartesianChart3.Series = viewM.Series;
                            cartesianChart3.XAxes = viewM.XAxes;
                            cartesianChart3.YAxes = viewM.YAxes; 
```

**Reply by CodeWithMichal** [reply]
Could you share reproducable sample with example data provided?

**Comment by SuperDaveOsbourne** [comment]
Did you ever get a working solution to this with volume bars below the candle sticks?
